### PR TITLE
[Vincenzo Spinello] pseudocontainers_task

### PR DESCRIPTION
**Pull request by vspinell**

### CODE REVIEW

File: pseudocontainers_task/student/src/impl.cpp [hunk 1]
```diff
@@ -1 +1,29 @@
 #include "impl.h"
+#include <bitset>
+
+bool isBiggerIntersection(const calendar& c1,const  calendar& c2, days& maxIntersectionCount) {
+	std::bitset<daysInMonth> bitFormatCalendar1(c1, 0, c1.size(),availableMarker, vacationMarker);
+	std::bitset<daysInMonth> bitFormatCalendar2(c2, 0, c2.size(),availableMarker, vacationMarker);
+	days IntersectionCount =  (bitFormatCalendar1 & bitFormatCalendar2).count();
+	if (IntersectionCount > maxIntersectionCount) {
```
Comment (smay1613) [inline pseudocontainers_task/student/src/impl.cpp]: std::max?

File: pseudocontainers_task/student/src/impl.cpp [hunk 2]
```diff
@@ -1 +1,29 @@
...
+	return false;
+}
+
+std::tuple<worker_id, worker_id, days> getMaxIntersection(const std::unordered_map<worker_id, calendar>& workerVacations) {
+	days maxIntersectionCount = 0;
+	std::tuple<worker_id, worker_id, days> maxIntersection;
+	if (!workerVacations.empty())
+	{
+		for (auto it = workerVacations.begin();  std::next(it) != workerVacations.end() ; ++it) {
+			for (auto it2 = std::next(it); it2 != workerVacations.end() ; ++it2) {
```
Comment (smay1613) [inline pseudocontainers_task/student/src/impl.cpp]: looks good, but please detalize why do you need  that next iterator
(do it with a good naming)

File: pseudocontainers_task/student/src/impl.cpp [hunk 3]
```diff
@@ -5,7 +5,7 @@ bool isBiggerIntersection(const calendar& c1,const  calendar& c2, days& maxInter
 	std::bitset<daysInMonth> bitFormatCalendar1(c1, 0, c1.size(),availableMarker, vacationMarker);
 	std::bitset<daysInMonth> bitFormatCalendar2(c2, 0, c2.size(),availableMarker, vacationMarker);
 	days IntersectionCount =  (bitFormatCalendar1 & bitFormatCalendar2).count();
-	if (IntersectionCount > maxIntersectionCount) {
+	if (std::max(IntersectionCount, maxIntersectionCount) != maxIntersectionCount) {
```
Comment (smay1613) [inline pseudocontainers_task/student/src/impl.cpp]: not this style, try just do:
maxIntersectionCount = std::max(IntersectionCount, maxInstersectionCout)

Comment (vspinell) [reply pseudocontainers_task/student/src/impl.cpp]: but I also want to return true. Just assigning will cause the condition to be always true

Comment (smay1613) [reply pseudocontainers_task/student/src/impl.cpp]: so please don't assign and check the condition in one function
consider to change design a little bit

**Pull request closed without merging**